File: ConditionalSelection.cpp
#include "stdafx.h"
#include "ConditionalSelection.h"
#include <iostream>

ConditionalSelection::ConditionalSelection() : Selection()
{
}

vector<vector<string>> ConditionalSelection::limitedSelection(vector<string> candidates, unsigned short dimension, unsigned short minimum[], unsigned short maximum[]) {
	final_elements = {};
	groupSelection(candidates, dimension);
	for (int i = 0; i < complete_group.size(); i++) {
		bool state = false;
		for (int j = 0; j < words.size(); j++) {
			// get 'words[j]' frequency/count in group
			int frequency = count(complete_group[i].begin(), complete_group[i].end(), words[j]);
			if (frequency >= minimum[j] && frequency <= maximum[j]) {
				state = true;
			}
			else {
				state = false;
				break;
			}
		}
		// skip if already in net
		if (state) {
			final_elements.push_back(complete_group[i]);
		}
	}
	return final_elements;
}

ConditionalSelection::~ConditionalSelection()
{
}

File: ConditionalSelection.py
from Selection import Choice

# define a class
class ConditionedChoice(Choice):

    def __init__(self):
        super().__init__()

    # pick only those groups that meet occurrence conditions
    def limitedSelection(self, candidates, size, minimum, maximum):
        self.final_elements = []

        self.groupSelection(candidates, size)
        for option in self.complete_group:
            state = False
            for j in range(len(self.words)):
                # get 'word[j]' frequency/count in group
                frequency = option.count(self.words[j])

                # Productive boundary check
                if frequency >= minimum[j] and frequency <= maximum[j]:
                    state = True
                else:
                    state = False
                    break

            # skip if already in net
            if state:
                self.final_elements.append(option)

        return self.final_elements
